### ISSUE TITLE
request parameters are duplicated when using a cursor paginator

### ISSUE THREAD

**Issue by girarda** [airbytehq/airbyte#23497]
## Current Behavior
tl;dr: request parameters are duplicated when using a cursor paginator

Some APIs return the URL of the next page of records as part of their response.

Example with a paginator setting a `limit` request parameter:

```
      paginator:
        type: DefaultPaginator
        page_token_option:
          type: RequestPath
        page_size_option:
          inject_into: request_parameter
          type: RequestOption
          field_name: limit
        pagination_strategy:
          type: CursorPagination
          cursor_value: '{{ response[''metadata''][''next''] }}'
          page_size: 1
````

part of the response will include the URL:
```
"next": "https://api.tempo.io/4/accounts?offset=5&limit=1",
```

On the next request, the `limit` request param will be duplicated if the URL returned by the API contains the request params (as is the case in the example above)

example request:
```
"next": "https://api.tempo.io/4/accounts?offset=5&limit=1&limit=1",
```

[slack thread](https://airbytehq-team.slack.com/archives/C027KKE4BCZ/p1677434219383379)

## Expected Behavior
* The simple retriever should deduplicate the request params before creating the prepared request

## Implementation details
* This can be down in the _fetch_next_page method
* urllib.urlparse can be used to extract the query params from the `path`
* This should be done in the SimpleRetriever
* `HttpStream` has the same potential for duplicate, but we should leave it out of scope to limit the blast radius of the change. We can create a follow up issue if needed.

## Open question
* How should we handle duplicate keys with different values?

## Acceptance criteria
- [ ] Request parameters are not duplicated when using the cursor pagination
- [ ] The CDK logs a warning when there are duplicate request params

**Comment by girarda** [comment]
grooming notes:
- on conflict: use the value that was declared in the manifest

**Comment by haithem-souala** [comment]
Hey,
Any ETA or workaround? this is blocking us from developing a new connector.

**Comment by girarda** [comment]
Noting this is also a [blocker for connectors on demand](https://airbytehq-team.slack.com/archives/C03TP6W8081/p1688998668728259).
